packages/uicore/src/components/ExpressionInput/ExpressionInput.tsx

```diff
@@ -209,9 +206,10 @@ export function ExpressionInput(props: ExpressionInputProps): React.ReactElement
               })
 
               const child = inputRef.current.childNodes[childIndex]
+
               if (child?.nodeType === Node.ELEMENT_NODE || child?.nodeType === Node.TEXT_NODE) {
-                const offset = childNodesTextLength[childIndex - 1] - position + 2
-                setCaret(child, offset)
+                const offset = position - childNodesTextLength[childIndex - 1] + 1
+                setCaret(child, offset, true)
```

Comment:
this true is to set the cursor position in case of complete element as this only get's called when we do an item select it will always be a complete element